sklearn/utils/_metadata_requests.py

```diff
@@ -0,0 +1,1201 @@
+
+    This decorator wraps around a router estimator's method, such as ``fit``,
+    and handles routing and and validation of the metadata passed to it.
+
+    .. versionadded:: 1.1
+
+    Parameters
+    ----------
+    func : method
+        A router estimator's method such as ``fit``.
+
+    Returns
+    -------
+    decorator : method
+        The estimator's method, after validation and routing of the metadata.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        sig = inspect.signature(func)
+        # Create the argument binding so we can determine what
+        # parameters are given what values
+        argument_binding = sig.bind(*args, **kwargs)
+        argument_map = argument_binding.arguments
+
+        # find parameter names which should be passed explicitly, and the kwarg param
+        # which can accept routed parameters.
+        func_params = list(sig.parameters.items())
+        params_to_pass = []
+        kwarg_param = None
+        for pname, param in func_params:
+            if param.kind == param.VAR_KEYWORD:
+                kwarg_param = pname
+                continue
+            params_to_pass.append(pname)
+
+        if kwarg_param is None:
+            raise TypeError("Method must accept kwargs for metadata routing to work.")
+
+        obj = argument_map["self"]
+        if not hasattr(obj, "get_metadata_routing"):
+            raise AttributeError(
+                f"This {repr(obj.__class__.__name__)} has not implemented the routing"
+                " method `get_metadata_routing`."
+            )
+        method_name = func.__name__
+        if method_name not in METHODS:
+            raise TypeError(
+                f"Can only route and process input on these methods: {METHODS}, "
+                f"while the current method is: {method_name}."
+            )
+
+        considered_params = {
+            key: value
+            for key, value in argument_map.items()
+            if key not in ("X", "y", "Y", "self")
+        }
+        # WWe pop the kwarg parameter from the params above, and then update it
+        # with the remaining values. This resembles the code snippets such as:
+        # if sample_weight is not None:
+        #     fit_params["sample_weight"] = sample_weight
+        if kwarg_param in considered_params:
+            routed_params = considered_params.pop(kwarg_param)
+            routed_params.update(considered_params)
+        else:
+            routed_params = considered_params
+
+        request_routing = metadata_router_factory(obj)
+        request_routing.validate_metadata(params=routed_params, method=method_name)
+        params = request_routing.get_params(params=routed_params, method=method_name)
+
+        # create a dict of other parameters and then add 'params' to them
+        explicit_params = {
+            key: value for key, value in argument_map.items() if key in params_to_pass
+        }
+        explicit_params["params"] = params
```

Comment:
should this rather be `routed_params`?

Reply:
this are the ones which are _consumed_ by the meta-estimator and hence passed explicitly, think: `metaestimator.fit(X, y, sample_weight, **fit_params)`.

Reply:
I'm talking about the key name `"params"` which is for the ones routed by this estimator.